code/game/objects/items/stacks/sheets/glass.dm

```diff
@@ -377,6 +377,64 @@ GLOBAL_LIST_INIT(plastitaniumglass_recipes, list(
 	force = 6
 	throwforce = 11
 	icon_state = "plasmalarge"
+	inhand_icon_state = "shard-plasma"
 	custom_materials = list(/datum/material/alloy/plasmaglass=MINERAL_MATERIAL_AMOUNT)
 	icon_prefix = "plasma"
 	weld_material = /obj/item/stack/sheet/plasmaglass
+
+/obj/item/shard/plasma/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/stack/sheet/cloth))
+		var/obj/item/stack/sheet/cloth/C = I
+		to_chat(user, span_notice("You begin to wrap the [C] around the [src]..."))
+		if(do_after(user, 70, target = src))
+			var/obj/item/knife/shiv/plasma/S = new /obj/item/knife/shiv/plasma
+			C.use(1)
+			to_chat(user, span_notice("You wrap the [C] around the [src] forming a makeshift weapon."))
+			remove_item_from_storage(src)
+			qdel(src)
+			user.put_in_hands(S)
+
+/obj/item/shard/titanium
+	name = "bright shard"
+	desc = "A nasty looking shard of titanium infused glass."
+	throwforce = 12
+	icon_state = "titaniumlarge"
+	inhand_icon_state = "shard-titanium"
+	custom_materials = list(/datum/material/alloy/titaniumglass=MINERAL_MATERIAL_AMOUNT)
+	icon_prefix = "titanium"
+	weld_material = /obj/item/stack/sheet/titaniumglass
+
+/obj/item/shard/titanium/attackby(obj/item/I, mob/user, params)
```

Comment:
these have the same issues but you'll just move this into `/obj/item/shard` behavior anyway